### PR TITLE
Benjy dev

### PR DESCRIPTION
**Pull request by BbodnerB**
Hey all,
So here is the pull request for merging the code we've made into the master.
As we discussed I tried to combine all the methods that are shared across the single model, joint and sequential training scripts. All these common files, including those that parse the arguments are under training_utils.py.

Note that it is possible now to switch between the models "--model torchvision_resnet18", or "--model cutout_resnet18"
It is also possible to select which layer to extract the gutouts from using the "--feature_module" and "--target_layer_names". Feel free to check out any additional arguments there. It's also possible to load weights explicitly into model a and model b.
I made a placeholder argument for changing the image resolution, though I haven't gotten to it yet.

Finally, I added the metrics we've discussed and verified they work properly by visualizing the accuracy in a small visualization script. I tested the code on the vm gpu and they all seem to run with and without the smoke_test flag.

Please add comments and suggested changes on the code wherever you think necessary, I'm sure there are things that don't make sense somewhere.

Thanks in advance!
Benjy

### CODE REVIEW

**Review by jiwonid — APPROVED** [review]

**Review by player-eric — APPROVED** [review]
Looks good to merge!
